### PR TITLE
Added API routes to reset forgotten password

### LINKED ISSUE #267: As a developer, I want to create a secure endpoint for the "Forgot password" functionality, so that unlogged-in users can securely change their passwords via a forgot password feature.
## Acceptance Criteria:  
- [ ] Endpoint that takes an email in the request body and sends an email to initiate password change is implemented.
- [ ] User identity is verified through the email before allowing password change.
- [ ] The endpoint integrates seamlessly with the Forgot Password feature.
- [ ] The password reset process adheres to security best practices, including secure token generation and email verification.

## Demo Steps:  
1. Demonstrate the initiation of a password change via the Forgot Password page using an email address.
2. Show the email received for the password reset and the subsequent password change process.
3. Verify the user identity and security measures throughout the process.

## Potential Risks:  
- Risk of security vulnerabilities in the email-based password reset process.
- Potential user confusion if the email process or password change steps are not clear and intuitive.
- Challenges in integrating the new endpoint with the existing Forgot Password system.

## Technical Notes:  
- Ensure the implementation of secure token generation for the password reset link in the email.
- Conduct comprehensive testing for the security and reliability of the new endpoint.
- Document the new API endpoint in Swagger under the 'Auth' section for straightforward frontend integration.

### PR DESCRIPTION
**Pull request by shadijiha**
Added API routes:

1. /profile/sentForgotPasswordEmail sends an forgot password email with a link to reset the password
2. /profile/resetPassword Resets the old password with a new one provided that you have the token from recovery email

NOTE:
Need to add the following to appsettings.json

```
    "Smtp": {
      "Username": "awaazo@outlook.com",
      "Password": "",  // <--- will provide password when asked
      "Port": "587",
      "Domain": "smtp-mail.outlook.com"
    },
```

Closes #267

### CODE REVIEW

**Review by mathieutrudeau — APPROVED** [review]

File: Backend/Backend/Controllers/ProfileController.cs [hunk 1]
```diff
@@ -153,6 +153,43 @@ public async Task<ActionResult> GetProfileAvatar()
...
+
+            await _profileService.SentForgotPasswordEmail(request.Email);
+            return Ok($"Email sent to {request.Email}");
+        }
+        catch (Exception e) {
+            this.LogErrorAPICall(_logger, e);
+            return BadRequest(e.Message);
+        }
+    }
+
```
Comment (BigSaladMan) [inline Backend/Backend/Controllers/ProfileController.cs:175]: Fairly certain you need the AllowAnonymous here as well right?

File: Backend/Backend/Services/ProfileService.cs [hunk 2]
```diff
@@ -194,4 +200,51 @@ public async Task<string> GetUserAvatarNameAsync(Guid userId)
         _db.Users.Update(user);
         await _db.SaveChangesAsync();
     }
+
+    public async Task SentForgotPasswordEmail(string requestEmail) {
+        // Verify that email exists
+        User? user = await _db.Users.Where(u => u.Email == requestEmail).FirstOrDefaultAsync();
```
Comment (BigSaladMan) [inline Backend/Backend/Services/ProfileService.cs:206]: this method should also clear up any old tokens attached to this email

**Review by BigSaladMan — APPROVED** [review]
Ill approve this, but there are some small changes that prob need to be made before merging.

**Comment by BigSaladMan** [comment]
another issue that could happen is that someone could just send a reset email request, then have the token linger on their account. Maybe not the end of the world, but something worth thinking about

File: Backend/Backend/Controllers/ProfileController.cs (hunk 1 above)
Comment (shadijiha) [reply Backend/Backend/Controllers/ProfileController.cs:175]: Done

**Comment by shadijiha** [comment]
> another issue that could happen is that someone could just send a reset email request, then have the token linger on their account since they dont actually finish the reset process. Maybe not the end of the world, but something worth thinking about

We can maybe add expiration on the tokens. Maybe like 10 min or something. What do you think?

**Pull request merged**